### DISCUSSION TITLE
Time between two CMP IR requests

### DISCUSSION THREAD

**Discussion by DavidGarriou** [Keyfactor/ejbca-ce#571] [General]
Hi,

I am wondering if there is any recommendation regarding waiting times between two IR requests from the same client (assuming the first one failed for a networking reason for example).
Does a client have to wait a minimum amount of time before trying a new IR request ?

Usually it is the responsibility of the network infrastructure, but is EJBCA protected against many incoming requests ?

Thanks a lot,
David

**Comment by primetomas** [comment]
Since you say IR, we are talking about CMP here I assume.
EJBCA itself does not do rate limiting on CMP. >1000req/second is totally fine, if your deployment infrastructure can handle it. So you're right, you need to work with the infra to tune DDoS protection depending on the load your specific environment can handle.
Parallel requests are fine are well. The CMP stress test that comes with clientToolBox does parallel requests. What you want to be careful with is doing parallel requests for the same End Entity as these will work on the same rows in the database. For parallel requests for different end entities it's full throttle.
See the note about concurrent transactions for the same end entity in the documentation.
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/protocols/cmp
